src/WebJobs.Script/Diagnostics/ValueStopwatch.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    // Based on: https://github.com/dotnet/aspnetcore/blob/master/src/Shared/ValueStopwatch/ValueStopwatch.cs
+    public struct ValueStopwatch
+    {
+        private static readonly double TimestampToTicks = TimeSpan.TicksPerSecond / (double)Stopwatch.Frequency;
+
+        private long _startTimestamp;
+
+        private ValueStopwatch(long startTimestamp)
+        {
+            _startTimestamp = startTimestamp;
+        }
+
+        public bool IsActive => _startTimestamp != 0;
```

Comment:
Question out of curiosity: Would there be a case where this prop returns `false`? (based on the comment on line 27)

Reply:
Yep - this is a `struct` to avoid the allocations, so a `default(ValueStopwatch)` would be false and appropriately go boom if ever used. Added in the test suite to illustrate :)